dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -96,22 +93,58 @@ static INT8 GetDriveNumber(PCUITEMID_CHILD pidl)
     return number < 26 ? number : -1;
 }
 
-template<class T> static T* GetDrivePath(PCUITEMID_CHILD pidl, T *Path)
+static INT8 GetDriveNumber(PCWSTR DrivePath)
+{
+    WORD number = (DrivePath[0] | 32) - 'a';
+    return number < 26 ? number : -1;
+}
+
+UINT g_IsFloppyCache = 0;
+
+static UINT GetCachedDriveType(int DrvNum)
+{
+    if (DrvNum < 0 || DrvNum >= 26)
+        return DRIVE_UNKNOWN;
+    if (g_IsFloppyCache & (1 << DrvNum))
+        return DRIVE_REMOVABLE;
+    WCHAR DrvPath[] = { LOWORD('A' + DrvNum), ':', '\\', '\0' };
+    return ::GetDriveTypeW(DrvPath);
+}
+
+static bool IsFloppyDrive(PCWSTR DrivePath)
+{
+    extern BOOL IsDriveFloppyW(LPCWSTR pszDriveRoot);
+    INT8 DrvNum = GetDriveNumber(DrivePath);
+    if (DrvNum >= 0)
```

Comment:
maybe:
```suggestion
    if (DrvNum < 0)
        return false;
```
and continue with less indentation level